src/blocks/content-helper/current-post-details/component.tsx

```diff
@@ -0,0 +1,309 @@
+						<div className={ 'bar-fill ' + key } key={ key }
+							title={ value.viewsPercentage + __( '%', 'wp-parsely' ) }
+							style={ { width: value.viewsPercentage + '%' } }>
+						</div>
+					);
+				} ) }
+			</div>
+
+			<table>
+				<thead>
+					<tr>{
+						Object.keys( data.referrers.types ).map( ( key ) => {
+							return <th key={ key }>{ key }</th>;
+						} ) }
+					</tr>
+				</thead>
+				<tbody>
+					<tr>{
+						Object.entries( data.referrers.types ).map( ( [ key, value ] ) => {
+							return <td key={ key }>{ ImpreciseNumber( value.views ) }</td>;
+						} ) }
+					</tr>
+				</tbody>
+			</table>
+		</div>
+	);
+}
+
+/**
+ * Outputs the "Top Referrers" section.
+ *
+ * @param {PostPerformanceData} data The data needed to populate the section.
+ */
+function TopReferrersSection( data: PostPerformanceData ) {
+	let totalViewsPercentage = 0;
+	return (
+		<div className="section top-referrers">
+			<table>
+				<thead>
+					<tr>
+						<th>{ __( 'Top Referrers', 'wp-parsely' ) }</th>
+						<th colSpan={ 2 }>{ __( 'Page Views', 'wp-parsely' ) }</th>
+					</tr>
+				</thead>
+				<tbody>{
+					Object.entries( data.referrers.top ).map( ( [ key, value ] ) => {
+						if ( key === 'totals' ) {
+							totalViewsPercentage = Number( value.viewsPercentage );
+							return null;
+						}
+
+						return (
+							<tr key={ key }>
+								<th title={ key }>{ key }</th>
+								<td title={ value.viewsPercentage + __( '%', 'wp-parsely' ) }>
+									<div className="percentage-bar" style={ { '--bar-fill': value.viewsPercentage + '%' } as React.CSSProperties }></div>
+								</td>
+								<td>{ ImpreciseNumber( value.views ) }</td>
+							</tr>
+						);
+					} )
+				}
+				</tbody>
+			</table>
+			<div>
+				{ totalViewsPercentage + __( '% of views came from top referrers.', 'wp-parsely' ) }
+			</div>
+		</div>
+	);
+}
+
+/**
+ * Outputs the "Actions" section.
+ *
+ * @param {PostPerformanceData} data The data needed to populate the section.
+ */
+function ActionsSection( data: PostPerformanceData ) {
+	return (
+		<div className="section actions">
+			<Button href={ data.url } rel="noopener" target="_blank" variant="secondary">{ __( 'Visit Post', 'wp-parsely' ) }</Button>
+			<Button href={ data.statsUrl } rel="noopener" target="_blank" variant="primary">{ __( 'View in Parse.ly', 'wp-parsely' ) }</Button>
+		</div>
+	);
+}
+
+/**
+ * Implements the "Imprecise Number" functionality of the Parse.ly dashboard.
+ *
+ * Note: This function is not made to process float numbers.
+ *
+ * @param {string} value          The number to process. It can be formatted.
+ * @param {number} fractionDigits The number of desired fraction digits.
+ * @param {string} glue           A string to put between the number and unit.
+ * @return {string} The number formatted as an imprecise number.
+ */
+function ImpreciseNumber( value: string, fractionDigits = 1, glue = '' ): string {
+	const number = parseInt( value.replace( /\D/g, '' ) );
```

Comment:
`parseInt()` should always include the second (radix) argument since it may not always default to `10`.

Reply:
Thanks for catching this!